src/zenml/cli/stack.py

```diff
@@ -141,6 +141,128 @@ def register_stack(
         cli_utils.declare(f"Stack '{stack_name}' successfully registered!")
 
 
+@stack.command("update", context_settings=dict(ignore_unknown_options=True))
+@click.argument("stack_name", type=str, required=True)
+@click.option(
+    "-m",
+    "--metadata-store",
+    "metadata_store_name",
+    help="Name of the new metadata store for this stack.",
+    type=str,
+    required=False,
+)
+@click.option(
+    "-a",
+    "--artifact-store",
+    "artifact_store_name",
+    help="Name of the new artifact store for this stack.",
+    type=str,
+    required=False,
+)
+@click.option(
+    "-o",
+    "--orchestrator",
+    "orchestrator_name",
+    help="Name of the new orchestrator for this stack.",
+    type=str,
+    required=False,
+)
+@click.option(
+    "-c",
+    "--container_registry",
+    "container_registry_name",
+    help="Name of the new container registry for this stack.",
+    type=str,
+    required=False,
+)
+@click.option(
+    "-s",
+    "--step_operator",
+    "step_operator_name",
+    help="Name of the new step operator for this stack.",
+    type=str,
+    required=False,
+)
+@click.option(
+    "-x",
+    "--secrets_manager",
+    "secrets_manager_name",
+    help="Name of the new secrets manager for this stack.",
+    type=str,
+    required=False,
+)
+def update_stack(
+    stack_name: str,
+    metadata_store_name: Optional[str] = None,
+    artifact_store_name: Optional[str] = None,
+    orchestrator_name: Optional[str] = None,
+    container_registry_name: Optional[str] = None,
+    step_operator_name: Optional[str] = None,
+    secrets_manager_name: Optional[str] = None,
+) -> None:
+    """Update a stack."""
+    # TODO: [MEDIUM] consider allowing stack to be renamed using this CLI command
```

Comment:
Do you think it makes sense to also allow removing a specific component from a stack? Right now that wouldn't be possible I think (and I can't think of an easy way to adapt this CLI method), but maybe that's not even a valid usecase?

Reply:
It probably does make sense, though I'm not sure what the CLI command would look like:

`zenml stack update -x`

Perhaps something like this where you pass in the flag for the command, but no corresponding stack component? Feels like this might be a bit unclear for the user, though. You might easily pass in a flag accidentally and then your component would get removed from the stack.

Maybe we could do the above behaviour, but have a confirmation prompt for whenever you're removing components?

Reply:
Yeah seems quite unclear and also hard to make the user aware of. 
I have another suggestion which I'm also not a huge fan of, but at least it would have a separate option and therefore allow us to write some help for it.
`zenml stack update --remove-component feature_store [--remove-component container_registry]`

Reply:
Another option (which I don't like) would be to have sub-sub-commands:

```shell
zenml stack update default delete -x secrets_manager
zenml stack update default add -x new_secrets_manager
```

But that'd then require multiple commands for everything.

Reply:
I went with a completely separate command for this:

`zenml stack remove-component -x -c`

etc